proc-macro/src/lib.rs

```diff
@@ -818,3 +827,27 @@ where
     map.insert(val);
     val
 }
+
+fn escape_utf8(string: &str, repr: &mut String) {
+    let mut chars = string.chars();
+    while let Some(ch) = chars.next() {
+        if ch == '\0' {
```

Comment:
I believe technically, this is unnecessary, as a CStr is not allowed to contain a null byte, but I'm not sure if it is worth it to change the implementation here.